pkg/cmd/server/kubernetes/master.go

```diff
@@ -356,7 +356,7 @@ func (c *MasterConfig) createSchedulerConfig() (*scheduler.Config, error) {
 	var configData []byte
 
 	// TODO make the rate limiter configurable
-	configFactory := factory.NewConfigFactory(c.KubeClient, kapi.DefaultSchedulerName, kapi.DefaultHardPodAffinitySymmetricWeight, kapi.DefaultFailureDomains)
+	configFactory := factory.NewConfigFactory(c.KubeClient, c.Options.SchedulerName, int(c.Options.HardPodAffinitySymmetricWeight), c.Options.FailureDomains)
```

Comment:
can we feed this from kube-scheduler `NewSchedulerServer`, `AddFlags`, and have a `schedulerArguments` in the config file instead? I don't see the number of options here diminishing and I'd rather be converging on kube's config